Android/src/org/droidplanner/android/fragments/SettingsFragment.java

```diff
@@ -261,17 +259,6 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 }
             });
         }
-
-        final CheckBoxPreference killSwitch = (CheckBoxPreference) findPreference(DroidPlannerPrefs.PREF_ENABLE_KILL_SWITCH);
```

Comment:
Most likely because the `Flight` activity is recreated when coming back from the `Settings` screen. But the `Settings` screen itself has access to the menu, and the preference toggle event will most likely not propagate to that menu.